Source/LuaBridge/detail/Result.h

```diff
@@ -45,6 +45,12 @@ struct Result
     {
         return m_ec.message();
     }
+    
+    void throw_on_error() const
+    {
+        if (!*this)
+            throw std::system_error(m_ec);
```

Comment:
What about if we throwing a runtime_error ? system_error is mostly for OS kind of errors

Reply:
Sounds good to me. For now i would disable the call when not building with exceptions. Maybe later we do what optional and expected are doing when accessing value() without a value, they abort (panic) the app.

Reply:
> But I don't understand why we would use std::error_code and then be reluctant to throw it with std::system_error.

I would maybe create a specific type for our exceptions. We can use a plain system_error, but maybe we should be able to differentiate when we are raising from luabridge, or we are catching an exception raised from from a std::thread or any other system_error throws by the standard library.

Reply:
I would be cool with that. FWIW, I believe `std::system_error` is a subclass of `runtime_error`, so anything catching `runtime_error` should also catch `system_error`.